server/src/main/java/com/objectcomputing/checkins/services/skills/SkillServicesImpl.java

```diff
@@ -0,0 +1,92 @@
+package com.objectcomputing.checkins.services.skills;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.validation.constraints.NotNull;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+@Singleton
+public class SkillServicesImpl implements SkillServices {
+
+    @Inject
+    private SkillRepository skillRepository;
+
+    public void setSkillRepository(SkillRepository skillsRepository) {
+        this.skillRepository = skillsRepository;
+    }
+
+    public Skill save(Skill skill) {
+        Skill newSkill = null;
+        if (skill != null) {
+            final String name = skill.getName();
+
+            if (skill.getSkillid() != null) {
+                throw new SkillBadArgException(String.format("Found unexpected id %s for skill, please try updating instead.",
+                        skill.getSkillid()));
+            } else if (skillRepository.findByName(skill.getName()).isPresent()) {
+                throw new SkillAlreadyExistsException(String.format("Member %s already has this skill %s", skill.getSkillid(), name));
```

Comment:
please can we reword the exception as it is Skill id and not the member id(member %s)

Reply:
done